gemrb/plugins/2DAImporter/2DAImporter.cpp

```diff
@@ -32,11 +32,8 @@ static bool StringCompKey(const std::string& str, TableMgr::key_t key)
 
 TableMgr::index_t p2DAImporter::npos = TableMgr::npos;
 
-bool p2DAImporter::Open(DataStream* str)
+bool p2DAImporter::Open(std::unique_ptr<DataStream> str)
 {
-	if (str == NULL) {
-		return false;
-	}
```

Comment:
I guess this is why.

If we dont want to accept `nullptr`, why use a pointer at all? Why not `Open(DataStream& str)`?

Reply:
That's the alternative I was thinking about, who's supposed to own that?

Reply:
yeah, might be something for later. We'd have to look at what is going on. Seems like the file shouldn't need to persist past the `Open` call, but I wouldn't be surprise if that isn't universally true. But if some plugin does need to take ownership, it can always `std::move` it from within `Open` itself.

Reply:
Technically of course. I just try to have signatures that tell me, as a caller, what to expect w. r. t. even potential ownership transfer.

These three things here have their individual interface class each and previously did `delete`, and I guess in the long term we will drop all raw pointers of `DataStream`. That's the rationale.

Reply:
it can also just take a `const&` and then copy if need be. Thats assuming there is even a need to own the stream at all.

Reply:
`const` ref, if I'm getting this right, won't work since the regular operations mutate the stream.

Regardless of the ownership concept, somebody just needs to delete it unambiguously.

Reply:
you're right. Of course because of all that it seems like `Open` taking ownership is indistinguishable from any other way an `Open` call taking nonconst reference might manipulate the stream, including just closing it. So I don't see why that would be more objectionable than what we have now with unchecked pointers. We could use `&&` if that felt better. Does anything expect to read (or do anything) with a stream after passing it to `Open`? Seems like `Open` is a logical owner of the stream to me, but I haven't looked closely.

Reply:
In _these_ cases: no, nothing after `Open`.

Reply:
no matter, what you have is good enough for now. if we care later we can figure it out.

Reply:
Yeah, I'll just merge now. Btw, do tests still don't work for you? All I had to do was rebuild with `-DUSE_TESTS=on`.

Reply:
they do now:

```bash
➜  build git:(master) ✗ ctest                                     
Test project /Users/brad/Projects/gemrb/build
    Start 1: Test_gemrb_core
1/4 Test #1: Test_gemrb_core ..................   Passed    0.62 sec
    Start 2: Test_2DAImporter
2/4 Test #2: Test_2DAImporter .................   Passed    0.30 sec
    Start 3: Test_IDSImporter
3/4 Test #3: Test_IDSImporter .................   Passed    0.12 sec
    Start 4: Test_INIImporter
4/4 Test #4: Test_INIImporter .................   Passed    0.13 sec

100% tests passed, 0 tests failed out of 4

Total Test time (real) =   1.18 sec
```